library/objective-c/EnvoyBridgeUtility.h

```diff
@@ -24,7 +24,7 @@ static inline envoy_data *toNativeDataPtr(NSData *data) {
 }
 
 static inline envoy_data toManagedNativeString(NSString *s) {
-  size_t length = s.length;
+  size_t length = [s lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
```

Comment:
I believe you need +1 for the `'\0'`

Reply:
Indeed.

Reply:
Turns out not. Added `testToManagedNativeStringUsingUTF8Chars` to validate.